mypy/test/data/check-inference.test

```diff
@@ -1415,13 +1415,16 @@ main: note: In member "f" of class "A":
 [case testAttributePartiallyInitializedToNoneWithMissingAnnotation]
 class A:
     def f(self) -> None:
-        self.x = None # E: Need type annotation for variable
+        self.x = None
```

Comment:
Also, it would be useful to have a similar test case with strict optional enabled.